MinVer.Lib/Version.cs

```diff
@@ -108,9 +108,9 @@ public int CompareTo(Version other)
                 ? this
                 : new Version(minMajorMinor.Major, minMajorMinor.Minor, default, new[] { "alpha", "0" }, this.height, this.buildMetadata);
 
-        public Version WithHeight(int height) =>
+        public Version WithHeight(int height, VersionPart autoIncrement) =>
             this.preReleaseIdentifiers.Count == 0 && height > 0
-                ? new Version(this.Major, this.Minor, this.Patch + 1, new[] { "alpha", "0" }, height, default)
+                ? new Version(this.Major + (autoIncrement == VersionPart.Major ? 1 : 0), this.Minor + (autoIncrement == VersionPart.Minor ? 1 : 0), this.Patch + (autoIncrement == VersionPart.Patch ? 1 : 0), new[] { "alpha", "0" }, height, default)
                 : new Version(this.Major, this.Minor, this.Patch, this.preReleaseIdentifiers, height, height == 0 ? this.buildMetadata : default);
 

```

Comment:
I don't find this particularly readable because of the encased conditional assignments...

Reply:
Yes, that's true. I always tend to over-inline. This should be split apart a bit.

Reply:
also, when auto incrementing, this does not reset the other (lower!?! what's the working here? :p ) segments to 0.

Reply:
Ah, good catch! 🤦‍♂️

Reply:
xunit told me once I wrote the tests :smile: